src/main/java/com/artipie/repo/OrgLayout.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.repo;
+
+import com.artipie.RepoConfig;
+import com.artipie.Settings;
+import com.artipie.SliceFromConfig;
+import com.artipie.asto.Key;
+import com.artipie.http.Slice;
+import com.artipie.http.async.AsyncSlice;
+import com.artipie.http.rs.RsStatus;
+import com.artipie.http.rs.RsWithBody;
+import com.artipie.http.rs.RsWithStatus;
+import com.artipie.http.slice.SliceSimple;
+import com.jcabi.log.Logger;
+import io.vertx.reactivex.core.Vertx;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+import org.cactoos.scalar.Unchecked;
+
+/**
+ * Flat repositories layout.
+ * @see RepoLayout
+ * @since 0.4
+ * @checkstyle ReturnCountCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class OrgLayout implements RepoLayout {
+
+    /**
+     * Repository path prefix.
+     */
+    private static final Pattern REPO_PREF =
+        Pattern.compile("/(?:[a-zA-Z0-9]+)/(?:[a-zA-Z0-9]+)(/.*)");
```

Comment:
The only difference between this pattern and the one at `FlatLayout.REPO_PREF` is the minimum amount of `/` symbols and parts in between them, 2 in case of `FlatLayout` and 3 in this case. This also could be decomposed. An example:
`Pattern REPO_PREF = new PathLevel(3)`

Reply:
@Sammers21 `Pattern` is a `final` class and can't be extended. Also, we may consider changing the format of first or second part here, so it'll require more changes than using pattern string. I'd keep it as is while the format is not established as a requirement.

Reply:
@g4s8, how about this?:
`Pattern REPO_PREF = Pattern.compile(new PathLevel(3))`